snooty/postprocess.py

```diff
@@ -976,7 +983,59 @@ def enter_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
             assert isinstance(argument, n.Reference)
             source = argument.refuri
 
-        self.openapi_pages[current_slug] = self.SourceData(source_type, source)
+        api_version = node.options.get("api_version", None)
+        resource_versions: Optional[List[str]] = None
+
+        # Fetch OpenAPI versioning data if options are present
+        if api_version and source == "cloud":
+            # Fetch latest git_hash for S3 versioning data
+            git_hash_response = requests.get("https://cloud.mongodb.com/version")
+            git_hash_response.raise_for_status()
+            git_hash = git_hash_response.text
+            response = requests.get(
+                f"https://mongodb-mms-prod-build-server.s3.amazonaws.com/openapi/{git_hash}-api-versions.json"
+            )
+            response.raise_for_status()
+            decoded_response = response.content.decode("utf-8")
+            data = yaml.safe_load(decoded_response)
+
+            # Malformed Version data
+            if (
+                not data
+                or "versions" not in data
+                or "major" not in data.get("versions")
+            ):
+                self.context.diagnostics[fileid_stack.current].append(
+                    InvalidOpenApiResponse(node.start[0])
+                )
+                return
+
+            version_data = data.get("versions")
+            major_versions = version_data.get("major")
+
+            # Version not present in version data
+            if api_version not in major_versions:
+                # Allows error-free diagnostic report
+                if not (
+                    isinstance(major_versions, list)
+                    and all(isinstance(mv, str) for mv in major_versions)
+                ):
+                    major_versions = []
+                self.context.diagnostics[fileid_stack.current].append(
+                    InvalidVersion(api_version, major_versions, node.start[0])
+                )
+                return
+
+            if api_version in version_data.keys():
+                resource_versions = version_data.get(api_version)
+            else:
+                resource_versions = []
```

Comment:
(Nit) We could potentially reduce this block to:

```
resource_versions = version_data.get(api_version, [])
```

I believe this handles the same key-checking logic + default value behavior. If this isn't the desired behavior, feel free to ignore!

Reply:
Totally